lib/msf/base/sessions/ldap.rb

```diff
@@ -139,4 +146,32 @@ def _interact_stream
     raise EOFError if (console.stopped? == true)
   end
 
+  def on_registered
+    start_keep_alive_loop
+  end
+
+  # Start a background thread for regularly sending a no-op command to keep the connection alive
+  def start_keep_alive_loop
+    self.keep_alive_thread = framework.threads.spawn('LDAP-shell-keepalive', false) do
+      keep_alive_timeout = 10 * 60 # 10 minutes
+      loop do
+        if client.last_interaction.nil?
+          remaining_sleep = keep_alive_timeout
+        else
+          remaining_sleep = keep_alive_timeout - (Time.now - client.last_interaction)
+        end
+        sleep(remaining_sleep)
+        if (Time.now - client.last_interaction) > keep_alive_timeout
```

Comment:
Bit of an edgecase, but is it possible to swap this out to use the process monotonic clock for its book keeping 🤞 

https://blog.dnsimple.com/2018/03/elapsed-time-with-ruby-the-right-way/

Reply:
Always good to do it right - thanks!